sentry_sdk/sessions.py

```diff
@@ -50,6 +54,43 @@ def auto_session_tracking(hub=None, session_mode="application"):
             hub.end_session()
 
 
+def is_auto_session_tracking_enabled_scope(scope):
+    # type: (sentry_sdk.Scope) -> bool
+    """
+    Utility function to find out if session tracking is enabled.
+
+    TODO: This uses the new scopes. When the Hub is removed, the function
+    is_auto_session_tracking_enabled should be removed and this function
+    should be renamed to is_auto_session_tracking_enabled.
+    """
+    should_track = scope._force_auto_session_tracking
+    if should_track is None:
+        client_options = sentry_sdk.get_client().options
+        should_track = client_options.get("auto_session_tracking", False)
+
+    return should_track
+
+
+@contextmanager
+def auto_session_tracking_scope(scope, session_mode="application"):
+    # type: (sentry_sdk.Scope, str) -> Generator[None, None, None]
+    """
+    Starts and stops a session automatically around a block.
+
+    TODO: This uses the new scopes. When the Hub is removed, the function
+    auto_session_tracking should be removed and this function
+    should be renamed to auto_session_tracking.
```

Comment:
Wouldn't doing this be a breaking change (if we wait until a minor)?

Reply:
No. Because of this import:  https://github.com/getsentry/sentry-python/blob/new-scopes-call-api/sentry_sdk/integrations/wsgi.py#L12 (the `*_scopes` functions are hidden from the user)

Reply:
How are the `*_scope` methods hidden from the user? I understand that if someone imports this function from `sentry_sdk.integrations.wsgi`, then they will import it as `auto_session_tracking`, but it is still possible for users to import this function from `sentry_sdk.sessions` as `auto_session_tracking_scope`.

What if we just rename this function to `_auto_session_tracking_scope` to indicate that it is not public API?

Reply:
Maybe I'm misunderstanding what you mean @szokeasaurusrex but for actually removing the Hub we'll need another major release right? We can rename the function safely then. The _deprecation_ of the Hub, these `*_scope` functions, and `hub` as argument for `auto_session_tracking` and `is_auto_session_tracking_enabled` can happen in a minor before that.